src/App.tsx

```diff
@@ -8,32 +9,42 @@ function getRandomName(): string {
 }
 
 export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
+  const [clockName, setColckName] = useState('Clock-0');
+  const [isClockVisible, setIsClockVisible] = useState(true);
+
+  useEffect(() => {
+    document.addEventListener('contextmenu', (event: MouseEvent) => {
+      event.preventDefault();
+      setIsClockVisible(false);
+    });
+
+    document.addEventListener('click', () => {
+      setIsClockVisible(true);
+    });
+
+    const timerId = window.setInterval(() => {
+      setColckName(getRandomName());
+    }, 3300);
+
+    return () => {
+      window.clearInterval(timerId);
+      document.removeEventListener('contextmenu', (event: MouseEvent) => {
```

Comment:
you return the same callback twice - once for adding eventListener - and the second one for removing. Let's create a separate method for that